packages/ng-query/src/lib/tests/query.spec.ts

```diff
@@ -0,0 +1,138 @@
+import { QueryProvider, UseQuery } from '../query';
+import { TestBed } from '@angular/core/testing';
+import { fetcher, flushPromises, simpleFetcher } from './test-utils';
+import { QueryClient } from '../query-client';
+import { QueryClient as QueryCore } from '@tanstack/query-core';
+import { QueryObserver } from '@tanstack/query-core';
+import { baseQuery } from '../base-query';
+import { subscribeSpyTo } from '@hirez_io/observer-spy';
+import { throwError } from 'rxjs';
+jest.mock('../base-query');
+
+describe('useQuery', () => {
+  let useQuery: UseQuery;
+  let client: QueryCore;
+
+  beforeEach(() => {
+    TestBed.configureTestingModule({});
+    useQuery = TestBed.inject(QueryProvider);
+    client = TestBed.inject(QueryClient);
+  });
+
+  it('should properly execute query', () => {
+    useQuery(['key0'], simpleFetcher, { staleTime: 1000 });
```

Comment:
What's the difference between simpleFetcher and simpleFetcher?

Reply:
you mean `simpleFetcher` and `fetcher` ? they are utils I copied from vue-query tests

Reply:
But simpleFetcher isn't a function that returns an observable, so it's irrelevant.

Reply:
I updated the functions to return observables instead of promises.